### DISCUSSION TITLE
anern inverter on esp32?

### DISCUSSION THREAD

**Discussion by piotr-ino** [syssi/esphome-smg-ii#52] [Ideas]
https://github.com/jonaubf/anern-inverter-monitoring/blob/main/anern_monitoring/anern_inverter/inverter.py
Hello, Would it be possible to make it work on esp32? I have such an inverter and I have been struggling for a month to connect it to HA. I have logs from datalogger communication with the inverter from wireshark

**Comment by syssi** [comment]
It looks like your inverter is `pipsolar` compatible.

**Comment by piotr-ino** [comment]
does not want to work

[logi.txt](https://github.com/syssi/esphome-smg-ii/files/13849833/logi.txt)

**Reply by piotr-ino** [reply]
[logs (1).txt](https://github.com/syssi/esphome-smg-ii/files/13849852/logs.1.txt)

**Reply by piotr-ino** [reply]
You can take a look at these logs new 1
[Nowy 1.txt](https://github.com/syssi/esphome-smg-ii/files/13850035/Nowy.1.txt)

**Comment by piotr-ino** [comment]
Sorry. Works. My 2 esp were damaged. Thanks for the great job

**Reply by syssi** [reply]
Are you using `pipsolar` now?

**Reply by piotr-ino** [reply]
Yes. And I bought the same components as you

**Reply by piotr-ino** [reply]
Hi. Is it possible to connect up to 1 esp8266 pipsolar + jk bms?

**Reply by syssi** [reply]
The baud rate of the JK-BMS is 115200 baud. You should use an hardware UART to talk to the BMS. In general it should be possible but it's a bit much work for a ESP8266 board. I recommend to use a ESP32 board instead or to use multiple ESPs.

**Reply by piotr-ino** [reply]
Cool. I have esp32. How to connect both repositories? Linux is Mars for me

**Reply by syssi** [reply]
Could you create an issue here? https://github.com/syssi/esphome-pipsolar

I will merge both YAMLs. This project isn't a good place to discuss `pipsolar` and `jk_bms` topics. ;-)
